.travis.yml

```diff
@@ -7,18 +7,30 @@ git:
   quiet: true
 
 stages:
+  - name: preflight
+    if: type = pull_request
   - name: tests
-    if: type IN (push, pull_request)
+    if: type = pull_request
   - name: images
     if: type = pull_request AND branch = master
 
 jobs:
   include:
-    - stage: tests
+    - stage: preflight
       name: "Unit Tests"
       python: 3.6
       before_script: pip install tox-travis
       script: tox
+    - stage: preflight
+      services: docker
+      name: "Images build"
+      before_script:
+        - cd deploy/images
+        # Pulling image for layer caching
+        - make -j 2 pull-base pull-inbox IMGTAG=latest
+      script:
+        - make -j 2
```

Comment:
Images created during this stage needs to be properly tagged because we don't wan't to overwrite the `latest` on docker hub.
If it is a PR they should have the tag PR-{$TRAVIS_PULL_REQUEST},
if it not a PR the tag sholud be {travis-$TRAVIS_BUILD_NUMBER}

Reply:
This is taken into account

Reply:
What's the reasoning behind having that check in the make file?
It is more logical that it is placed in the travis file since it is only then it will have an effect and all travis related stuff will be in one place.

Reply:
There is no check in the Makefile.

Reply:
https://github.com/EGA-archive/LocalEGA/blob/e0e5f9ee5c7e18cd4fbb1c8e89a77832c7c122b9/deploy/images/Makefile#L11:L18